python/ray/tune/tests/test_ray_trial_executor.py

```diff
@@ -33,26 +35,45 @@ def tearDown(self):
         ray.shutdown()
         self.cluster.shutdown()
 
-    # no autoscaler case, resource is not sufficient. Raise error.
-    def testRaiseErrorNoAutoscaler(self):
+    # no autoscaler case, resource is not sufficient. Log warning for now.
+    @patch.object(ray.tune.trial_executor.logger, "warning")
+    def testRaiseErrorNoAutoscaler(self, mocked_warn):
+        class FailureInjectorCallback(Callback):
+            """Adds random failure injection to the TrialExecutor."""
+
+            def __init__(self, steps=5):
+                self._step = 0
+                self.steps = steps
+
+            def on_step_begin(self, iteration, trials, **info):
+                self._step += 1
+                if self._step >= self.steps:
+                    raise RuntimeError
```

Comment:
why do we need this?

Reply:
This is a test to break upon a certain threshold. It's fine to leave it.